### DISCUSSION TITLE
1.x VS 2.x

### DISCUSSION THREAD

**Discussion by kazuser** [pemistahl/lingua-py#224] [Q&A] [answered]
![image](https://github.com/pemistahl/lingua-py/assets/3335273/e3a2eb4b-ad94-418a-88df-973800bd25c3)

So... what is the difference? :)

**Comment by pemistahl** [comment, marked as answer]
No offense, but... sometimes I really wonder why people are not willing to read the [documentation](https://github.com/pemistahl/lingua-py?tab=readme-ov-file#3-a-short-history-of-this-library) thoroughly.

> Starting from version 2.0.0, the pure Python implementation was replaced with compiled Python bindings to the native [Rust implementation](https://github.com/pemistahl/lingua-rs) of Lingua. This decision has led to both quick performance and a small memory footprint of less than 1 GB. The pure Python implementation is still available in a [separate branch](https://github.com/pemistahl/lingua-py/tree/pure-python-impl) in this repository and will be kept up-to-date in subsequent 1.* releases. Both 1.* and 2.* versions will remain available on the Python package index (PyPI).

So use 1.x whenever you need a pure Python implementation, e.g. when you are in an environment that does not support native extensions. Debugging is also a lot easier in pure Python code.
